### PR TITLE
Add Binary Search for java

### PR DESCRIPTION
**Pull request by ricktjwong**
- Added binary search logic for java, handling integer arrays. Function returns the index of integer in the array
- Added pom.xml to allow running of java code
- Added tests
- Edited README.md to include instructions of how to run code

### CODE REVIEW

**Comment by nicholette** [comment]
Thanks for the PR @ricktjwong!

**Comment by ricktjwong** [comment]
> Thanks for the PR @ricktjwong!

More work for @tumtubs :)

**Comment by Bertrand31** [comment]
There's ReasonML in this PR :sweat_smile: And a conflict

File: java/src/main/java/Main.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import search.BinarySearch;
+
+import java.io.IOException;
+
+public class Main {
+    public static void main(String[] args) {
```
Comment (tumtubs) [inline java/src/main/java/Main.java]: @ricktjwong I don't think we should be including entry points or test cases. The purpose of LYANL is to introduce the language through well known algorithms. Maybe in the future we can have proper project setups to teach the ecosystems and how to deploy (this is actually more useful than the language itself IMO). But for now we should stick to algorithm implementation only.

File: java/src/test/java/search/BinarySearchTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package search;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class BinarySearchTest {
```
Comment (tumtubs) [inline java/src/test/java/search/BinarySearchTest.java]: Same here. You can keep this in your forked repo for now.

File: java/src/main/java/search/BinarySearch.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        if (!present) {
+            throw new IOException("Integer not present in array");
+        } else {
+            int low = 0;
+            int high = intArray.length - 1;
+            Arrays.sort(intArray);
+            return recursiveSearch(low, high, intArray, searchInt);
+        }
+    }
+    public int recursiveSearch(int low, int high, int[] intArray, int searchInt) {
```
Comment (tumtubs) [inline java/src/main/java/search/BinarySearch.java]: This could be extended to support more collections

**Pull request closed without merging**